class/frontend-dashboard.php

```diff
@@ -66,25 +64,114 @@ function post_listing( $post_type ) {
             echo '<div class="success">' . __( 'Post Deleted', 'wpuf' ) . '</div>';
         }
 
-        $args = array(
-            'author'         => get_current_user_id(),
-            'post_status'    => array('draft', 'future', 'pending', 'publish', 'private'),
-            'post_type'      => $post_type,
-            'posts_per_page' => wpuf_get_option( 'per_page', 'wpuf_dashboard', 10 ),
-            'paged'          => $pagenum
-        );
+        $post_type_array = array();
+        $args = array();
+        $available_posttypes = get_post_types();
+        $cond = 0;
+
+        if ( $form_id !== 'off' && $post_type === 'default' ) {

```

Comment:
Instead of making all these arrays from scratch, create a base `$args` array. Then based on the params, add new items to the array. Finally, the query will be built and execute the query.